Add runtime API endpoint for dynamic tracing log level control

sntntn opened the pull request:
This PR introduces the ability to change tracing log levels at runtime via an HTTP API endpoint.  
It leverages `tracing_subscriber::reload` layers for dynamic updates without restarting the node.

### Changes
- Added `POST /eth/v2/debug/tracing/log_level` endpoint
- Implemented `TracingHandle` wrapper for type-safe log level control
- Threaded `TracingHandle` through application layers (main → runtime → HTTP API)
- Integrated request handler in `standard.rs`
- Updated routing with new debug endpoint


### Example usage
``` API
curl -X POST http://localhost:5052/eth/v2/debug/tracing/log_level \
  -H "Content-Type: application/json" \
  -d '{
    "target": "fork_choice_control",
    "level": "debug"
  }'
```
> https://www.youtube.com/embed/c8RrdKiOQno
> In this video example, the log level for the `fork_choice_control` module is changed  
from `info` → `debug` → back to `info` using a simple **POST** request.

sntntn committed:
use reload layers for dynamic log level change + test crate database changing level to debug

Relates to #16

sntntn committed:
add runtime tracing log level API endpoint

- Implement POST /eth/v2/debug/tracing/log_level to dynamically change log levels for specific crates without application restart
- Implement TracingHandle wrapper for type safety
- Thread tracing handle through application layers (main -> runtime -> HTTP API)
- Add handler in standard.rs with JSON reqest/response
- Add route in eth_v2_debug_routes

Relates to #17

sntntn committed:
cargo fmt

Tumas reviewed (commented):
check the output of `/scripts/ci/clippy.bash`

sntntn committed:
refactor(tracing): make TracingHandle optional and reduce type complexity

Make tracing_handle optional in HttpApi (mirroring metrics) to simplify testing,
and factor out TracingHandle’s complex type into a type alias to resolve Clippy warnings.

Relates to #20 and #21

sntntn committed:
apply clippy suggestions in http_api

sntntn commented:
> check the output of `/scripts/ci/clippy.bash`

<img width="881" height="945" alt="Screenshot from 2025-09-30 20-21-05" src="https://github.com/user-attachments/assets/bc583ae6-4aaa-4042-8618-8849b976ad7b" />
